### ISSUE TITLE
inconsistent hatch and border color in barh in matplotlib 2.1.0

### ISSUE THREAD

**Issue by amueller** [matplotlib/matplotlib#9808]
### Bug report

It seems edgecolor and hatch is not applied to all bars in a barh in 2.1.0

**Code for reproduction**

```python
bars = plt.barh(y=range(5), width=5, height=.6,  edgecolor='k', linewidth=5)
```

**Actual outcome**

![image](https://user-images.githubusercontent.com/449558/32957688-751f685a-cb8a-11e7-8cee-9b25f5594516.png)

**Expected outcome**

edge on all bars

**Matplotlib version**

  * Operating system: 
  * Matplotlib version: 2.1.0
  * Matplotlib backend: module://ipykernel.pylab.backend_inline

Other interesting phenomenon:
combining edgecolor and hatch makes hatch only be applied to the first bar:
```python
bars = plt.barh(y=range(5), width=5, height=.6, hatch="//", edgecolor='k', linewidth=5)
```
![image](https://user-images.githubusercontent.com/449558/32957828-bdbde58c-cb8a-11e7-8cd7-d280bb5ec10f.png)
but hatch on itself works as expected:
```python
bars = plt.barh(y=range(5), width=5, height=.6, hatch="//")
```
![image](https://user-images.githubusercontent.com/449558/32957851-d31e1d20-cb8a-11e7-9918-cfcf4dc511c4.png)

(the linewidth is not required to reproduce anywhere, it's just to make the effect more visible)

**Comment by jklymak** [comment]
Thanks for the bug report!  The snippet above works on master and the fix will be in 2.1.1.  See #9351

**Comment by jklymak** [comment]
Please feel free to re-open if 2.1.1 doesn't fix it for you
